source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -415,6 +422,32 @@ bool JsonTranscoderFilter::readToBuffer(Protobuf::io::ZeroCopyInputStream& strea
   return false;
 }
 
+void JsonTranscoderFilter::buildResponseFromHttpBodyOutput(Http::HeaderMap& response_headers,
+                                                           Buffer::Instance& data) {
+  try {
+    const Json::ObjectSharedPtr http_body = Json::Factory::loadFromString(data.toString());
+    const std::string decoded_body = Base64::decode(http_body->getString("data"));
+
+    data.drain(data.length());
+    data.add(decoded_body);
+
+    response_headers.insertContentType().value(http_body->getString("contentType"));
+    response_headers.insertContentLength().value(decoded_body.size());
+  } catch (const Json::Exception& e) {
+    ENVOY_LOG(debug, "Failed to parse output of '{}'. e.what(): {}", method_->full_name(),
+              e.what());
+  }
+}
+
+bool JsonTranscoderFilter::hasHttpBodyAsOutputType() {
+  absl::string_view output_type = method_->output_type()->full_name();
+  absl::string_view http_body_type = google::api::HttpBody::descriptor()->full_name();
+  if (output_type.length() != http_body_type.length()) {
+    return false;
+  }
+  return StringUtil::startsWith(output_type.data(), http_body_type.data(), true);
```

Comment:
why startsWith but not equal here? You should use `absl::StartsWith` for `string_view`s.

Reply:
I have the length comparison previously. And since the comparison needs to be case-sensitive?

Reply:
Will use the equal operator then.